main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/microkit/command"
+	microserver "github.com/giantswarm/microkit/server"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/viper"
+
+	"github.com/giantswarm/auto-oncall/flag"
+	"github.com/giantswarm/auto-oncall/server"
+	"github.com/giantswarm/auto-oncall/service"
+)
+
+var (
+	description string     = "This is the webhook handler application for creating opsgenie routing rules from github deployment events."
+	f           *flag.Flag = flag.New()
+	gitCommit   string     = "n/a"
+	name        string     = "auto-oncall"
+	source      string     = "https://github.com/giantswarm/auto-oncall"
+)
+
+func main() {
+	var err error
+
+	// Create a new logger which is used by all packages.
+	var newLogger micrologger.Logger
+	{
+		c := micrologger.Config{}
+
+		newLogger, err = micrologger.New(c)
+		if err != nil {
+			panic(fmt.Sprintf("%#v", err))
+		}
+	}
+
+	// We define a server factory to create the custom server once all command
+	// line flags are parsed and all microservice configuration is storted out.
+	newServerFactory := func(v *viper.Viper) microserver.Server {
+		var newService *service.Service
+		{
+			c := service.Config{
+				Logger: newLogger,
+
+				Description: description,
+				Flag:        f,
+				GitCommit:   gitCommit,
+				Name:        name,
+				Source:      source,
+				Viper:       v,
+			}
+
+			newService, err = service.New(c)
+			if err != nil {
+				panic(fmt.Sprintf("%#v", err))
+			}
+		}
+
+		var newServer microserver.Server
+		{
+			c := server.Config{
+				Flag:    f,
+				Logger:  newLogger,
+				Service: newService,
+				Viper:   v,
+
+				ProjectName: name,
+			}
+
+			newServer, err = server.New(c)
+			if err != nil {
+				panic(fmt.Sprintf("%#v", err))
+			}
+		}
+
+		return newServer
+	}
+
+	// Create a new microkit command which manages our custom microservice.
+	var newCommand command.Command
+	{
+		c := command.Config{
+			Logger:        newLogger,
+			ServerFactory: newServerFactory,
+
+			Description: description,
+			GitCommit:   gitCommit,
+			Name:        name,
+			Source:      source,
+		}
+
+		newCommand, err = command.New(c)
+		if err != nil {
+			panic(fmt.Sprintf("%#v", err))
+		}
+	}
+
+	daemonCommand := newCommand.DaemonCommand().CobraCommand()
+
+	daemonCommand.PersistentFlags().String(f.Service.Oncall.OpsgenieToken, "", "Opsgenie API token.")
+	daemonCommand.PersistentFlags().String(f.Service.Oncall.Users, "", "github_id:opsgenie_id mapppings, separated by comma.")
```

Comment:
Would it make sense to use `map[string]string` for `Users`? See https://godoc.org/github.com/spf13/pflag#FlagSet.GetStringToString and https://github.com/spf13/pflag/pull/133 so it should work `--service.oncall.users=github_id0=opsgenie_id0,github_id1=opsgenie_id1,...` - then one can write user mappings as a YAML map in installations file and do something similar to what we did with string slices here: https://github.com/giantswarm/aws-operator/blob/master/helm/aws-operator-chart/templates/03-configmap.yaml#L18

Later when querying for value, I think you would just do: `viper.GetStringMapString(f.Service.Oncall.Users)` and get a map of users.

Reply:
I'm using map[string]string for Users on the service level. But with flag this magic didn't work for me. It just fails to automatically dispatch that flag into map[string]string